test/unit/test_logger.py

```diff
@@ -126,3 +126,28 @@ def test_cyan_text():
     x = '{}{}{}'.format(colorama.Fore.CYAN, 'foo', colorama.Style.RESET_ALL)
 
     assert x == logger.cyan_text('foo')
+
+
+def test_markup_detection_pycolors_0(monkeypatch):
+    monkeypatch.setenv('PY_COLORS', '0')
+    assert not logger.should_do_markup()
+
+
+def test_markup_detection_pycolors_1(monkeypatch):
+    monkeypatch.setenv('PY_COLORS', '1')
+    assert logger.should_do_markup()
+
+
+def test_markup_detection_is_tty(monkeypatch, mocker):
+    monkeypatch.delenv('PY_COLORS', raising=False)
+    monkeypatch.setenv('TERM', 'xterm')
+    stdout_mock = mocker.patch("sys.stdout")
```

Comment:
There's no need in patching stdlib objects if you decide what to pass to the tested function anyway?